semantics/src/main/scala/aqua/semantics/rules/mangler/ManglerInterpreter.scala

```diff
@@ -1,22 +1,20 @@
 package aqua.semantics.rules.mangler
 
+import aqua.mangler.ManglerState
+
 import cats.data.State
 import monocle.Lens
-import monocle.macros.GenLens
 
 class ManglerInterpreter[X](using
   lens: Lens[X, ManglerState]
 ) extends ManglerAlgebra[State[X, *]] {
 
   override def rename(name: String): State[X, String] =
     for {
-      s <- get
-      newName = LazyList
-        .from(0)
-        .map(i => s"$name-$i")
-        .dropWhile(s.isForbidden)
-        .head
-      _ <- modify(_.forbid(newName))
+      state <- get
+      result = state.forbidAndRename(name)
+      (newState, newName) = result
+      _ <- modify(_ => newState)
```

Comment:
Maybe rewrite `modify` according to use case?
Smth like
```
private def apply[A](f: ManglerState => (ManglerState, A)): State[X, A] =
    State.apply(lens.modifyF(f andThen (_.swap)) andThen (_.swap))
```
(have to swap because of functor instances)
and then
`newName <- apply(_.forbidAndRename)`